src/EbmlElement.cpp

```diff
@@ -195,26 +195,22 @@ EbmlElement * EbmlElement::FindNextID(IOCallback & DataStream, const EbmlCallbac
     } while (_SizeLength == 0);
   }
 
-  const auto PossibleID = EbmlId(EbmlId::FromBuffer(PossibleId.data(), PossibleID_Length));
-  auto Result = [&] {
-    if (PossibleID != EBML_INFO_ID(ClassInfos))
-    {
+  auto Result = [&]() -> EbmlElement * {
+    auto pID = EbmlId(EbmlId::FromBuffer(PossibleId.data(), PossibleID_Length));
+    if (pID != EBML_INFO_ID(ClassInfos)) {
       if (SizeFound == SizeUnknown)
-        return static_cast<EbmlElement *>(nullptr);
-      return static_cast<EbmlElement *>(new EbmlDummy(PossibleID));
+        return nullptr;
+      return new EbmlDummy(pID);
     }
     if (SizeFound != SizeUnknown && MaxDataSize < SizeFound)
-      return static_cast<EbmlElement *>(nullptr);
+      return nullptr;
     // check if the size is not all 1s
     if (SizeFound == SizeUnknown && !ClassInfos.CanHaveInfiniteSize())
-      return static_cast<EbmlElement *>(nullptr);
+      return nullptr;
     return &EBML_INFO_CREATE(ClassInfos);
   }();
 
-  if (Result == nullptr)
-    return nullptr;
-
-  if (!Result->SizeIsValid(SizeFound)) {
+  if (!Result || !Result->SizeIsValid(SizeFound)) {
```

Comment:
I'm not a fan of calling delete on nullptr when we know it's nullptr. Some compiler might complain it's wasted call. Also this is unrelated to the commit.

Reply:
Calling `delete` on a `nullptr` is completely normal and expected. See [cppref](https://en.cppreference.com/w/cpp/language/delete); `nullptr` handling is totally baked into the specs of `delete`. It's fine.

Reply:
I know it's fine, but that doesn't mean it's not a wasted call.

Anyway that line is gone in #273 as the size is validated before.

Reply:
You could just as easily argue that checking for `nullptr`, which is done inside `delete` as well, is wasted code that'll blow up the code size unnecessarily & therefore affect code locality & cache coherence negatively. Without actual measurements one point of view is just as right (and just as wrong) as the other.

Reply:
Honestly this should be using shares_ptr. Although that’s probably a deep change